content/languages/cpp/authoring/index.md

```diff
@@ -0,0 +1,378 @@
+
+template<typename ExpressionType, typename MessageSupplierType>
+static void That(const char* actual, const ExpressionType& expression, const MessageSupplierType& message_supplier, const char* file_name = "", int line_number = 0);
+```
+
+`message_supplier` is a callable compatible with a function accepting no arguments and returning `std::string`. It can be a function, a functor, a lambda expression, or an instance of any type supporting above requirements.
+
+For more details and examples of custom assertion messages, see `[Example test suite](#example_test_suite)` below, or visit the `[Snowhouse reference](/languages/cpp/snowhouse/)` page.
+
+#### Stringizers
+
+In its default configuration, Snowhouse can produce confusing assertion messages when `expected` and `actual` values are of a type it cannot stringify:
+
+```text
+  does_not_pretty_print_type_without_stringizer
+    Expected: equal to [ [unsupported type], [unsupported type] ]
+    Actual: [ [unsupported type], [unsupported type] ]
+```
+
+To rectify this issue in your tests, you can make such types suitable for stringification in a way described in the dedicated documentation article on [custom Snowhouse stringizers](/languages/cpp/igloo/stringizers/).
+
+
+### Random utilities
+
+Until C++11, the most common way of producing random values was the `rand` function. However, it has a set of problems: it needs to be properly seeded, and it is difficult to produce values outside of `0...RAND_MAX` range. Since C++11, the standard library offers a set of functionalities which are designed to produce random values of different types, from various ranges, and with better distribution when compared to `rand`. Unfortunately, the API presented in the `random` header seems to be confusing and difficult to use and accompanied by a few misconceptions, and authors either use it incorrectly or resort to (not) good, old `rand`. However, working with `random` turns out to be not that difficult:
+
+```cpp
+//use random_device only as a seeder
+std::random_device seeder;
+
+//create one PRNG which will be used to pick values
+//from (potentially many) distributions
+std::mt19937 engine{ seeder() };
+
+//a set of distributions for every type or range you are going to need in your tests
+std::uniform_int_distribution<    int> rand_number     {  1,  100 };
+std::uniform_int_distribution< size_t> rand_length     { 20,  100 };
+std::uniform_real_distribution<double> rand_coefficient{ -1,    1 };
+std::uniform_int_distribution<   char> rand_letter     { 'a', 'z' };
+std::uniform_int_distribution<   char> rand_bool       {   0,   1 };
+//etc.
+
+//for convenience, distributions can be bound with PRNG
+auto gen_length = std::bind(rand_length, engine);
+auto gen_letter = std::bind(rand_letter, engine);
+
+std::string input;
+//input string generator: string of random length, composed of random letters
+size_t input_length = gen_length(); //use length generator
+for(int i=0; i<input_length; ++i) {  
```

Comment:
`size_t i`? Comparing signed with unsigned will probably generate a warning. (I guess `int input_length` is OK too for kata-scale sizes if it appears in an existing translation, but for best practices it should be `size_t`.)

Reply:
🤦 
```suggestion
for(size_t i=0; i<input_length; ++i) {  
```